changelogs/fragments/4038-fix-and-rework-gitlb-project-variable.yml

```diff
@@ -0,0 +1,4 @@
+bugfixes:
+  - gitlab_project_variable - allow to set same variable name under different environment scopes (https://github.com/ansible-collections/community.general/issues/4038).
```

Comment:
While this new feature contains bugfixes, it is primarily a new feature. It needs a `minor_changes` entry as well (mentioning the new option).